pkg/crc/api/api_http_test.go

```diff
@@ -0,0 +1,71 @@
+package api
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	cmdConfig "github.com/code-ready/crc/cmd/crc/cmd/config"
+	"github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/machine/fakemachine"
+	"github.com/code-ready/crc/pkg/crc/preflight"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHTTPApi(t *testing.T) {
+	machine := fakemachine.NewClient()
+	config := setupNewInMemoryConfig()
+
+	ts := httptest.NewServer(NewMux(config, machine))
+
+	client := http.DefaultClient
+	res, err := client.Get(fmt.Sprintf("%s%s", ts.URL, "/version"))
```

Comment:
Extract this client to a real pkg that we will support like pkg/api/client

Reply:
Done, PTAL again :)